### ISSUE TITLE
DateSegmentInfo.monthDates showing incorrect first date for visible month

### ISSUE THREAD

**Issue by ronbuencamino** [patchthecode/JTAppleCalendar#748]
IMPORTANT - BEFORE CREATING YOUR ISSUE PLEASE READ THE FOLLOWING:

1. PLEASE STATE THE VERSION NUMBER you are using.
7.1.5
2. Are you using the latest version of JTAppleCalendar? Latest version is currently 7.1.5
YES
3. PROVIDE ENOUGH IMPORMATION SO I CAN RECREATE THE PROBLEM.

I'm doing something like this:
`self.setCalendarEvents(visibleDates: self.calendarController.visibleDates())`
`func setCalendarEvents(visibleDates: DateSegmentInfo) {
        let first = visibleDates.monthDates.first
        print("first: \(first)")
        
    }`

The visible month dates should be the month of March 2018. The indates start Feb 25 and go until Feb 28. However, when I print out 'first' above I get Feb 25 as appearing in monthdates  instead of March 1. Is there a reason an indate would be appearing in the monthdate grouping?

**Comment by patchthecode** [comment]
can you show me a screen shot of the month that is visible?

**Comment by ronbuencamino** [comment]
![simulator screen shot - iphone 8 - 2018-04-02 at 13 18 32](https://user-images.githubusercontent.com/27229172/38214105-73dfc9d6-3678-11e8-97df-f3a67edbf3c6.png)

When I print out visibleDates.monthDates from didScrollToDateSegmentWith I get the following data:

[(date: 2018-02-25 08:00:00 +0000, indexPath: [0, 28]), (date: 2018-02-26 08:00:00 +0000, indexPath: [0, 29]), (date: 2018-02-27 08:00:00 +0000, indexPath: [0, 30]), (date: 2018-02-28 08:00:00 +0000, indexPath: [0, 31]), (date: 2018-03-01 08:00:00 +0000, indexPath: [1, 4]), (date: 2018-03-02 08:00:00 +0000, indexPath: [1, 5]), (date: 2018-03-03 08:00:00 +0000, indexPath: [1, 6]), (date: 2018-03-04 08:00:00 +0000, indexPath: [1, 7]), (date: 2018-03-05 08:00:00 +0000, indexPath: [1, 8]), (date: 2018-03-06 08:00:00 +0000, indexPath: [1, 9]), (date: 2018-03-07 08:00:00 +0000, indexPath: [1, 10]), (date: 2018-03-08 08:00:00 +0000, indexPath: [1, 11]), (date: 2018-03-09 08:00:00 +0000, indexPath: [1, 12]), (date: 2018-03-10 08:00:00 +0000, indexPath: [1, 13]), (date: 2018-03-11 08:00:00 +0000, indexPath: [1, 14]), (date: 2018-03-12 07:00:00 +0000, indexPath: [1, 15]), (date: 2018-03-13 07:00:00 +0000, indexPath: [1, 16]), (date: 2018-03-14 07:00:00 +0000, indexPath: [1, 17]), (date: 2018-03-15 07:00:00 +0000, indexPath: [1, 18]), (date: 2018-03-16 07:00:00 +0000, indexPath: [1, 19]), (date: 2018-03-17 07:00:00 +0000, indexPath: [1, 20]), (date: 2018-03-18 07:00:00 +0000, indexPath: [1, 21]), (date: 2018-03-19 07:00:00 +0000, indexPath: [1, 22]), (date: 2018-03-20 07:00:00 +0000, indexPath: [1, 23]), (date: 2018-03-21 07:00:00 +0000, indexPath: [1, 24]), (date: 2018-03-22 07:00:00 +0000, indexPath: [1, 25]), (date: 2018-03-23 07:00:00 +0000, indexPath: [1, 26]), (date: 2018-03-24 07:00:00 +0000, indexPath: [1, 27]), (date: 2018-03-25 07:00:00 +0000, indexPath: [1, 28]), (date: 2018-03-26 07:00:00 +0000, indexPath: [1, 29]), (date: 2018-03-27 07:00:00 +0000, indexPath: [1, 30]), (date: 2018-03-28 07:00:00 +0000, indexPath: [1, 31]), (date: 2018-03-29 07:00:00 +0000, indexPath: [1, 32]), (date: 2018-03-30 07:00:00 +0000, indexPath: [1, 33]), (date: 2018-03-31 07:00:00 +0000, indexPath: [1, 34])]

**Comment by patchthecode** [comment]
now that i have seen your screen, this is expected.
You did `calendarView.visibleDates`
And according to the screenshot, The dates from feb-25 to feb-28 are visible on the screen.

Maybe if you let me know what you are trying to do then i can help

**Comment by ronbuencamino** [comment]
Sorry, I meant to say I printed calendarView.visibleDates.monthDates and was expecting 3/1 to be the first date, not 2/26

-RB

**Comment by patchthecode** [comment]
what will be visible first, would be the **actual** visible dates.

So let me ask you, when you ran that command, was Febuary's date visible on the screen?

**Comment by ronbuencamino** [comment]
Yes, that’s correct. I hid feb dates because it was an indate and I didn’t want to display it. However, it was my assumption for monthDates that it would have been the first of the actual month that’s being presented, in this case 3/1 since it’s for the month of March. 

That said, what’s the best way to get the monthDates for the current month segment that’s being displayed? 

-RB

**Comment by patchthecode** [comment]
Here is the thing based on the screen shot, both months are being presented on the screen.
Yes, March takes up a majority portion of the screen, but February is still presented on the screen (25th to 28th). Since 2 months are displayed on the screen, the library cannot know which month you want displayed, therefore we leave that job up to the developer.

If you let me know what you're trying to do then i can better help.


A ugly fix would simple be, check the monthDates and sort by month. If FebDates.count > MarchDates.count then you can know that feb is on the screen. If MarchDates.count > FebDates.count, then probably march is on the screen. But there is no way for me to tell

1. May i ask what you need the month dates for?
2. May I know what is your `calendarView.scrollMode`?

**Comment by ronbuencamino** [comment]
1. I’m doing some background fetching for events for a given month. I’m using visibleMonths.monthDates.first as my anchor when fetching. As a workaround, I grab the last day of the month and calculate the start from there. It seems to be accurate enough. 

2. Scroll mode is vertical 

After you’ve had explained it, that’s fine with me. I have a workaround, I just wanted to raise attention as the values for monthDates seemed incorrect, as even though two months were being displayed I had imagined the feb dates to be indates instead. 

Thanks for a great library!
